backend/src/main/java/com/phaete/backend/forage/service/ForageMapItemService.java

```diff
@@ -0,0 +1,87 @@
+package com.phaete.backend.forage.service;
+
+import com.phaete.backend.forage.model.*;
+import com.phaete.backend.forage.repository.ForageMapItemRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+@RequestMapping("/api/forageMapItem")
+public class ForageMapItemService {
+
+	private final ForageMapItemRepository forageMapItemRepository;
+	private final ConverterService converterService;
+
+	public ForageMapItemService(
+			ForageMapItemRepository forageMapItemRepository,
+			ConverterService converterService
+	) {
+		this.forageMapItemRepository = forageMapItemRepository;
+		this.converterService = converterService;
+	}
+
+	/**
+	 * Creates a new forage map item and saves it to the database.
+	 *
+	 * @param forageMapItemDTO the forage map item to be created
+	 * @return the newly created forage map item
+	 */
+	public ForageMapItemDTO createForageMapItem(ForageMapItemDTO forageMapItemDTO) {
+		return converterService.toDTO(
+				forageMapItemRepository.save(
+						converterService.fromDTO(forageMapItemDTO)
+				)
+		);
+	}
+
+
+	/**
+	 * Retrieves all forage map items from the database.
+	 * <p>
+	 * @return a list of all forage map items that contain valid customMarkers and forageWikiItems
+	 */
+	public List<ForageMapItem> findAllForageMapItems() {
+		Map<Boolean, List<ForageMapItem>> forageMapItemMap = forageMapItemRepository.findAll()
+				.stream()
+				.collect(
+						Collectors.partitioningBy(
+								forageMapItem -> (
+										forageMapItem.getForageWikiItem() != null &&
+										forageMapItem.getCustomMarker() != null
+								)
+						)
+				);
+		return forageMapItemMap.get(true);
+	}
+
```

Comment:
Why partioningBy? Wouldnt filter have the same effect? It's easier to understand.

Reply:
This is in preparation for the next feature implementation.
I want to have a logger log all the "faulty" ForageMapItems to fix them. If I ignore them, the DB will eventually be full of faulty items and just take up unnecessary space

Reply:
And the check if an entry is valid, can also be extracted in an extra method.